core/tee/tee_fs_common.c

```diff
@@ -391,14 +391,15 @@ static int read_and_decrypt_file(int fd,
 	if (!ciphertext) {
 		EMSG("Failed to allocate file data buffer, size=%zd",
 				file_size);
-		return -1;
+		res = FD_MAP_ERR(TEE_ERROR_OUT_OF_MEMORY);
```

Comment:
I think we can still return -1 and add another function to get the error reason like this:

static TEE_Result tee_fs_result = TEE_SUCCESS;
TEE_Result last_tee_fs_result(void) 
{
        return tee_fs_result;
}